sonic-net/sonic-buildimage
# Pddf_support_dell_platforms: Add PDDF support for s52xx dell platforms.

#### Why I did it
Add PDDF support for the dell s52xx platforms.

#### How I did it
Add PDDF configuration files, scripts and python files.

#### How to verify it
To switch to PDDF mode run the below command and wait for the system to be ready.
pddf_util.py -d switch-pddf

Run the PDDF commnads 
systemctl | grep -i pddf
pddf_fanutil direction
pddf_fanutil getspeed
pddf_fanutil numfans
pddf_fanutil status
pddf_psuutil mfrinfo
pddf_psuutil numpsus
pddf_psuutil seninfo
pddf_psuutil status 
pddf_thermalutil gettemp
pddf_thermalutil numthermals
pddf_ledutil getstatusled SYS_LED
pddf_ledutil getstatusled FAN_LED
pddf_ledutil getstatusled FANTRAY1_LED
pddf_ledutil getstatusled FANTRAY2_LED
pddf_ledutil getstatusled FANTRAY3_LED
pddf_ledutil getstatusled FANTRAY4_LED

Note: To switch back to non PDDF mode check help on pddf_util.py -help


Unit Test results:
[s5232f_pddf_logs.txt](https://github.com/sonic-net/sonic-buildimage/files/15225499/s5232f_pddf_logs.txt)
[s5248f_pddf_logs.txt](https://github.com/sonic-net/sonic-buildimage/files/15225500/s5248f_pddf_logs.txt)
[s5296_pddf_logs.txt](https://github.com/sonic-net/sonic-buildimage/files/15225501/s5296_pddf_logs.txt)
[s5212f_pddf_logs.txt](https://github.com/sonic-net/sonic-buildimage/files/15225502/s5212f_pddf_logs.txt)
[s5224f_pddf_logs.txt](https://github.com/sonic-net/sonic-buildimage/files/15225503/s5224f_pddf_logs.txt)

#### A picture of a cute animal (not mandatory but encouraged)
